test_conformance/atomics/test_atomics.cpp

```diff
@@ -200,6 +200,10 @@ int test_atomic_function(cl_device_id deviceID, cl_context context, cl_command_q
         error = clGetKernelWorkGroupInfo( kernel, deviceID, CL_KERNEL_WORK_GROUP_SIZE, sizeof( workSize ), &workSize, NULL );
         test_error( error, "Unable to obtain max work group size for device and kernel combo" );
 
+        // Limit workSize to avoid extremely large local buffer size and slow
+        // run.
+        if (testFns.NumResultsFn && workSize > 65536) workSize = 65536;
```

Comment:
Why are you also testing `testFns.NumResultsFn` is non-zero?

Reply:
@kpet, thanks for comments. Checking the context of the program, I found that `testFns.NumResultsFn` is used to calculate the local buffer size, and only when it's set, local buffer is required. So if it's not set, we don't need the clamp the `workSize` to limit local buffer usage.

Reply:
Would it work to always clamp the size? It is always queried and, assuming I understand correctly, this would help readability I think.

Reply:
Done. Thanks!

Reply:
Looks good, thanks for the changes (and the patience!)